test/unit/annotation_spec.js

```diff
@@ -3857,6 +3858,61 @@ describe("annotation", function () {
     );
   });
 
+  describe("FreeTextAnnotation", () => {
+    it("should create an new FreeText annotation", async () => {
+      partialEvaluator.xref = new XRefMock([]);
+      const task = new WorkerTask("test FreeText creation");
+      const data = await AnnotationFactory.saveNewAnnotations(
+        partialEvaluator,
+        task,
+        [
+          {
+            annotationType: AnnotationEditorType.FREETEXT,
+            rect: [12, 34, 56, 78],
+            fontSize: 10,
+            color: [0, 0, 0],
+            value: "Hello PDF.js World!",
+          },
+        ]
+      );
+
+      const base = data.annotations[0].data.replace(/\(D:\d+\)/, "(date)");
+      expect(base).toEqual(
+        "1 0 obj\n" +
+          "<< /Type /Annot /Subtype /FreeText /CreationDate (date) " +
+          "/Rect [12 34 56 78] /DA (/Helv 10 Tf 0 g) /Contents (Hello PDF.js World!) " +
+          "/F 4 /Border [0 0 0] /Rotate 0 /AP << /N 2 0 R>>>>\n" +
+          "endobj\n"
+      );
+
+      const font = data.dependencies[0].data;
+      expect(font).toEqual(
+        "0 0 obj\n" +
```

Comment:
Wait, how did this actually happen since object `0 0` should always be free in a valid XRef table?

Reply:
It's because of:
https://github.com/mozilla/pdf.js/blob/7761d2bf954f36abffef959913f34a5e6533949f/src/core/xref.js#L50

So it shouldn't be an issue with a real pdf.
Do you prefer I add a `|| 1`:
```js
      this._newRefNum = this.entries.length || 1;
```
?

Reply:
> So it shouldn't be an issue with a real pdf.

That's good to know, thank you!

> Do you prefer I add a `|| 1`:
> ```js
>       this._newRefNum = this.entries.length || 1;
> ```

Yes, please do that, since I think it's a good idea to ensure that we don't generate "bogus" data even if it's *only during the unit-tests.